packages/bpk-component-input/src/BpkInput.tsx

```diff
@@ -156,12 +150,13 @@ class BpkInput extends Component<Props, State> {
             tabIndex="-1"
             label={clearButtonLabel || ''}
             onMouseDown={onMouseDown}
-            onClick={(e) => {
+            onClick={(e: SyntheticEvent<HTMLButtonElement>) => {
               if (ref) {
                 ref.focus();
               }
               if (onClear) {
-                e.target.name = name;
+                const target = e.target as HTMLButtonElement;
```

Comment:
We've just recently stripped out all type assertions in banana. There's quite a few articles like this as to why you should avoid using them - https://netbasal.com/why-you-should-avoid-using-type-assertion-in-typescript-e44e8217cae5

In this instance could you do this instead?

```
              if (onClear) {
                if (e.target instanceof HTMLButtonElement) {
                  const { target } = e;
                  target.name = name;
                }
                onClear(e);
                this.setState({ persistClearButton: false });
              }

Reply:
Ah interesting, the point stated in the article makes sense.
I think in this case though it won't really make a difference as the event target will always be an HTMLButtonElement and neither the type nor the actual target are something we can change?